src/Popover/index.js

```diff
@@ -23,9 +23,17 @@ const Popover = ({
   disableAutoPlacement = false,
   matchTriggerWidth = false,
   testId,
+  closeOnSelect = false,
 }) => {
   const [isOpen, setIsOpen] = useState(false);
   const focusTrapRef = useFocusTrap(isOpen);
+  const popoverContent = closeOnSelect ? React.cloneElement(content, {
```

Comment:
The `Popover` is generic and could be a menu or _any_ arbitrary content.

Could we call this prop `closeOnContentClick` to be more clear?